src/passes/MergeBlocks.cpp

```diff
@@ -264,7 +272,16 @@ struct MergeBlocks : public WalkerPass<PostWalker<MergeBlocks>> {
   }
 
   void visitSelect(Select* curr) {
-    optimize(curr, curr->condition, optimize(curr, curr->ifFalse, optimize(curr, curr->ifTrue), &curr->ifTrue), &curr->ifTrue, &curr->ifFalse);
+    Block* outer = nullptr;
+    outer = optimize(curr, curr->ifTrue, outer);
+    if (EffectAnalyzer(getPassOptions(), curr->ifTrue).hasSideEffects()) return;
+    outer = optimize(curr, curr->ifFalse, outer);
+    if (EffectAnalyzer(getPassOptions(), curr->ifFalse).hasSideEffects()) return;
+    /* . */ optimize(curr, curr->condition, outer);
```

Comment:
This line seems weird.
... This function seems *really* weird.

So I like splitting the line.
I like aligning the `optimize`s. I think that `/* . */` is a confusing way to do that. Better would be just whitespace, but then because it comes on a line after an `if`, it looks like the body of the `if`. Although with the comment it still looks like that.
The comment does do a good job of sticking out and saying "hey this is unusual," so that worked.
The `if` statements are copy-pasted from `optimize`'s internals.

So, I propose:
```c++
Block* outer = nullptr;

outer = optimize(curr, curr->ifTrue,    outer);
outer = optimize(curr, curr->ifFalse,   outer, &(curr->ifTrue));
        optimize(curr, curr->condition, outer, &(curr->ifTrue), &(curr->ifFalse));
```